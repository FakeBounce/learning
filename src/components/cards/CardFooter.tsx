import { Trans } from '@lingui/macro';
import { Box } from '@mui/material';
import ActionButton from '@src/components/lms/ActionButton';
import { memo } from 'react';

interface CardFooterProps {
  isLoading: boolean;
  cancelAction: () => void;
}

function CardFooter({ isLoading, cancelAction }: CardFooterProps) {
  return (
    <Box display="flex" gap={2}>
      <ActionButton actionType="cancel" loading={isLoading} onClick={cancelAction}>
        <Trans>Annuler</Trans>
      </ActionButton>
      <ActionButton type="submit" role="submit" loading={isLoading}>
        <Trans>Enregistrer</Trans>
      </ActionButton>
    </Box>
  );
}

export default memo(CardFooter);
