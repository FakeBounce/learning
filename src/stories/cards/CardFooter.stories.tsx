import { Meta, StoryObj } from '@storybook/react';
import CardFooter from '@src/components/cards/CardFooter';
import { fn } from '@storybook/test';

const meta: Meta = {
  title: 'Example/CardFooter',
  component: CardFooter,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;
type Story = StoryObj<typeof CardFooter>;

const CardFooterTemplate: Story = {
  render: (args: any) => {
    return <CardFooter {...args} />;
  }
};

export const Default = {
  ...CardFooterTemplate,
  args: {
    isLoading: false,
    cancelAction: fn(() => console.log('Cancel'))
  }
};

export const Loading = {
  ...CardFooterTemplate,
  args: {
    isLoading: true,
    cancelAction: fn(() => console.log('Cancel'))
  }
};
