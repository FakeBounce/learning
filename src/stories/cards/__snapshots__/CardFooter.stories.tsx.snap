// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/CardFooter Default smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1nvwnbn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="0"
          type="button"
          id=":r0:"
  >
    Annuler
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="0"
          type="submit"
          id=":r1:"
          role="submit"
  >
    Enregistrer
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;

exports[`Example/CardFooter Loading smoke-test 1`] = `
<div class="MuiBox-root css-10egq61">
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1nvwnbn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="-1"
          type="button"
          disabled
          id=":r2:"
  >
    <div class="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorCenter css-1iyu4y0-MuiLoadingButton-loadingIndicator">
      <span class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
            role="progressbar"
            aria-labelledby=":r2:"
            style="width: 16px; height: 16px;"
      >
        <svg class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
             viewbox="22 22 44 44"
        >
          <circle class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  r="20.2"
                  fill="none"
                  stroke-width="3.6"
          >
          </circle>
        </svg>
      </span>
    </div>
    Annuler
  </button>
  <button class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-121cmne-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          tabindex="-1"
          type="submit"
          disabled
          id=":r3:"
          role="submit"
  >
    <div class="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorCenter css-1iyu4y0-MuiLoadingButton-loadingIndicator">
      <span class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
            role="progressbar"
            aria-labelledby=":r3:"
            style="width: 16px; height: 16px;"
      >
        <svg class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
             viewbox="22 22 44 44"
        >
          <circle class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  r="20.2"
                  fill="none"
                  stroke-width="3.6"
          >
          </circle>
        </svg>
      </span>
    </div>
    Enregistrer
  </button>
</div>
`;
